src/renderer/src/routes/pages/Send/index.tsx

```diff
@@ -1,114 +1,237 @@
+      return null
+    }
+
+    const service = bsAggregatorRef.current.getBlockchainByName(selectedAccount.blockchain)
+
+    const key = await window.api.decryptBasedEncryptedSecret(selectedAccount.encryptedKey, encryptedPasswordRef.current)
+    if (!key) throw new Error(t('error.decryptKey'))
+    const serviceAccount = service.generateAccountFromKey(key)
+    return {
+      serviceAccount: serviceAccount,
+      service: service,
+      token: selectedToken,
+    }
+  }, [bsAggregatorRef, encryptedPasswordRef, selectedAccount, selectedAmount, selectedRecipient, selectedToken, t])
+
+  const populateTotalFee = useCallback(async () => {
+    const sendService = await populateSendService()
+    if (!sendService) {
+      return
+    }
+    if (!isCalculableFee(sendService.service)) {
+      return
+    }
+    const fee = await sendService.service.calculateTransferFee({
+      senderAccount: sendService.serviceAccount,
+      intent: {
+        receiverAddress: selectedRecipient,
+        tokenHash: sendService.token.token.hash,
+        amount: selectedAmount.toString(),
+        tokenDecimals: sendService.token.token.decimals,
+      },
+    })
+    setTotalFee(fee)
+  }, [populateSendService, selectedAmount, selectedRecipient])
+
+  useEffect(() => {
+    populateTotalFee()
+  }, [populateTotalFee])
+
+  const showTransactionStatus = (hash: string) => {
+    window.open(DoraHelper.buildTransactionUrl(hash), '_blank')
+  }
+
+  const handleSendToken = async () => {
+    const sendService = await populateSendService()
+    if (!sendService) {
+      return
+    }
+    if (!isCalculableFee(sendService.service)) {
+      return
     }
-    if (currentStep === step) {
-      return 'text-neon'
+    try {
+      const transactionHash = await sendService.service.transfer({
+        senderAccount: sendService.serviceAccount,
+        intent: {
+          receiverAddress: selectedRecipient,
+          tokenHash: sendService.token.token.hash,
+          amount: selectedAmount.toString(),
+          tokenDecimals: sendService.token.token.decimals,
+        },
+      })
+      if (!transactionHash) {
+        showErrorModal()
+      } else {
+        modalNavigate('success', {
+          state: {
+            heading: t('title'),
+            headingIcon: <TbStepOut />,
+            subtitle: t('sendSuccess.title'),
+            content: (
+              <Fragment>
+                <div className="flex w-full justify-center mt-16">
+                  <Button
+                    className="w-[11rem]"
+                    type="submit"
+                    label={t('viewStatus')}
+                    leftIcon={<TbEye />}
+                    onClick={() => {
+                      showTransactionStatus('0x6aad308c63b4d7afa5c41ecb01db836a7b833c7c57a01f9ac3ed934113c3361b')
+                    }}
+                  />
+                </div>
+              </Fragment>
+            ),
+          },
+        })
+      }
```

Comment:
Create a new function for this called showSuccessModal, I think it will make the code more readable